WordPress/gutenberg
# [ New block ] - Create a Generic Avatar block for Authors

## What

A Proof of Concept of having a generic Avatar block for anywhere we could use it on a WordPress Site.

It would deprecate `Comment Author Avatar`, and also it would avoid the creation of one specific block for `Post Author Avatar` (#38265 can be closed if we merge this one)

Partial for #35596
Also closes #31962, #35693, #38456

## Why

Using this approach we increase the logic of the code, but also, simplify the user experience. The user may not know the differences between a post or a comment author avatar. Just want an avatar to be displayed in some part of the site.

Also, with this case, we add the possibility of adding avatars outside of the post-comment context. For example, creating a "review product pattern", a "list of authors pattern", etc.

## How

In this PR we create a new Core block that:
- Will automatically fetch the avatar image depending on the context (Comment > Post > Selected by user).
- Has `duotone`, `border`, `alignment`, `spacing` and `color` options.
- Will show a selected user in case the editor selects that option.

## Screenshots
![Screenshot 2022-03-08 at 17 54 37](https://user-images.githubusercontent.com/37012961/157286613-9f622ee5-332f-4aac-b4f1-95273474fba7.png)

![Screenshot 2022-03-08 at 17 54 21](https://user-images.githubusercontent.com/37012961/157286657-2884213b-cdf2-4fdd-9a8f-b02907a39d21.png)


## Video demo

[Video Demo](https://www.loom.com/share/204f98601f924d9f9ecf16bad13b6a9c)

## How to test

Create some users with avatars. A couple of posts. And then:

- Check that the Avatar is working on post edition
- Check that it is working on a page edition
- Check that it is working on Site Editor Templates
- Check that it is working on Frontend
- Check that custom styling is applied
